keras_cv/models/object_detection/yolox/yolox.py

```diff
@@ -0,0 +1,1036 @@
+            uses a `NonMaxSuppression` operation for box pruning.
+        feature_pyramid: (Optional) A `keras.Model` representing a feature pyramid
+            network (FPN).  The feature pyramid network is called on the outputs of the
+            `backbone`.  The KerasCV default backbones return three outputs in a list,
+            but custom backbones may be written and used with custom feature pyramid
+            networks.  If not provided, a default feature pyramid network is produced
+            by the library.  The default feature pyramid network is compatible with all
+            standard keras_cv backbones.
+        evaluate_train_time_metrics: (Optional) whether or not to evaluate metrics
+            passed in `compile()` inside of the `train_step()`.  This is NOT
+            recommended, as it dramatically reduces performance due to the synchronous
+            label decoding and COCO metric evaluation.  For example, on a single GPU on
+            the PascalVOC dataset epoch time goes from 3 minutes to 30 minutes with this
+            set to `True`. Defaults to `False`.
+        name: (Optional) the name to be passed to the model. Defaults to `"YoloX"`.
+    """
+
+    def __init__(
+        self,
+        classes,
+        bounding_box_format,
+        phi,
+        backbone,
+        include_rescaling=None,
+        backbone_weights=None,
+        label_encoder=None,
+        prediction_decoder=None,
+        feature_pyramid=None,
+        evaluate_train_time_metrics=False,
+        name="YoloX",
+        **kwargs,
+    ):
+        label_encoder = label_encoder or YoloXLabelEncoder(
+            bounding_box_format=bounding_box_format
+        )
+
+        super().__init__(
+            bounding_box_format=bounding_box_format,
+            label_encoder=label_encoder,
+            name=name,
+            **kwargs,
+        )
+        self.evaluate_train_time_metrics = evaluate_train_time_metrics
+
+        self.depth_multiplier = DEPTH_MULTIPLIERS[phi]
+        self.width_multiplier = WIDTH_MULTIPLIERS[phi]
+
+        self.bounding_box_format = bounding_box_format
+        self.classes = classes
+        self.backbone = _parse_backbone(
+            backbone,
+            include_rescaling,
+            backbone_weights,
+            self.depth_multiplier,
+            self.width_multiplier,
+        )
+
+        self.prediction_decoder = prediction_decoder or DecodePredictions(
+            bounding_box_format=bounding_box_format,
+            classes=classes,
+        )
+
+        self.feature_pyramid = feature_pyramid or YoloXPAFPN(
+            depth_multiplier=self.depth_multiplier,
+            width_multiplier=self.width_multiplier,
+        )
+        self.yolox_head = YoloXHead(classes, width_multiplier=self.width_multiplier)
+
+        self._metrics_bounding_box_format = None
+        self.loss_metric = tf.keras.metrics.Mean(name="loss")
```

Comment:
Can this metrics be passed through compile?

Reply:
The loss metrics, you mean? 
These were added following the [RetinaNet implementation](https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/object_detection/retina_net/retina_net.py). Since YoloX has separate box loss, classification loss, and objectness loss, it made sense to follow the RetinaNet implementation style.

Reply:
I see. Some users might not want those metrics though. It sounds like maybe we should think about dict loss / dict metrics if we want a more friendly UX for adding metrics?
@LukeWood

Reply:
is the concern the metrics are too noisy?  

Separating the losses is super helpful for debugging.  we should definitely offer it; and I'm not sure its worth the API bloat to offer a way to disable that.

Reply:
Yes, we already provide those loss-metrics as Keras level, so adding more metrics here for logging losses seems weird